ocpmodels/datasets/ase_datasets.py

```diff
@@ -0,0 +1,355 @@
+class AseReadMultiStructureDataset(AseAtomsDataset):
+    """
+    This Dataset can read multiple structures from each file using ase.io.read.
+    The disadvantage is that all files must be read at startup.
+    This is a significant cost for large datasets.
+
+    This is intended for small-scale testing and demonstrations of OCP.
+    Larger datasets are better served by the efficiency of other dataset types
+    such as LMDB.
+
+    For a full list of ASE-readable filetypes, see
+    https://wiki.fysik.dtu.dk/ase/ase/io/io.html
+
+    args:
+        config (dict):
+            src (str): The source folder that contains your ASE-readable files
+
+            pattern (str): Filepath matching each file you want to read
+                    ex. "*.traj", "*.xyz"
+                    search recursively with two wildcards: "**/POSCAR" or "**/*.cif"
+
+            a2g_args (dict): Keyword arguments for ocp.preprocessing.AtomsToGraphs()
+                    default options will work for most users
+
+                    If you are using this for a training dataset, set
+                    "r_energy":True and/or "r_forces":True as appropriate
+                    In that case, energy/forces must be in the files you read (ex. OUTCAR)
+
+            ase_read_args (dict): Keyword arguments for ase.io.read()
+
+            keep_in_memory (bool): Store data in memory. This helps avoid random reads if you need
+                    to iterate over a dataset many times (e.g. training for many epochs).
+                    Not recommended for large datasets.
+
+            use_tqdm (bool): Use TQDM progress bar when initializing dataset
+
+            atoms_transform_args (dict): Additional keyword arguments for the atoms_transform callable
+
+            transform_args (dict): Additional keyword arguments for the transform callable
+
+        atoms_transform (callable, optional): Additional preprocessing function applied to the Atoms
+            object. Useful for applying tags.
+
+        transform (callable, optional): Additional preprocessing function for the Data object
+    """
+
+    def __init__(self, config, transform=None, atoms_transform=apply_one_tags):
+        super(AseReadMultiStructureDataset, self).__init__(
+            config, transform, atoms_transform
+        )
+        self.ase_read_args = self.config.get("ase_read_args", {})
+        if not hasattr(self.ase_read_args, "index"):
+            self.ase_read_args["index"] = ":"
+
+        self.path = Path(self.config["src"])
+        if self.path.is_file():
+            raise Exception("The specified src is not a directory")
+        self.filenames = sorted(self.path.glob(f'{self.config["pattern"]}'))
+
+        self.id = []
+        if self.config.get("use_tqdm", True):
+            self.filenames = tqdm(self.filenames)
+        for filename in self.filenames:
+            try:
+                structures = ase.io.read(filename, **self.ase_read_args)
```

Comment:
Looks like if `keep_in_memory` is False, all we're doing by reading the data upfront is indexing.

Would it make sense to ask users to provide an index file along with the ase-readable files? That way this won't be any slower than the single structure ase dataset.